addons/mail/controllers/discuss.py

```diff
@@ -70,14 +70,20 @@ def discuss_channel_welcome(self, channel_id):
     def mail_channel_partner_avatar_128(self, channel_id, partner_id, **kwargs):
         channel_partner_sudo = request.env['mail.channel.partner']._get_as_sudo_from_request_or_raise(request=request, channel_id=int(channel_id))
         if not channel_partner_sudo.env['mail.channel.partner'].search([('channel_id', '=', int(channel_id)), ('partner_id', '=', int(partner_id))], limit=1):
-            raise NotFound()
+            if request.env.user.share:
+                placeholder = channel_partner_sudo.env['res.partner']._avatar_get_placeholder()
```

Comment:
should call the method on the actual partner of `partner_id` to be completely correct (this would take into account company partner etc)